plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/SwitchVisibility/index.tsx

```diff
@@ -0,0 +1,308 @@
+
+  const handleStyleChange = useCallback((style: UIStyles) => {
+    selectStyle(style);
+  }, []);
+
+  const styleOptions = (
+    <Menu
+      items={Object.keys(uiStyles).map(key => {
+        return {
+          key: key,
+          label: (
+            <p style={{ margin: 0 }} onClick={() => handleStyleChange(key as UIStyles)}>
+              {uiStyles[key]}
+            </p>
+          ),
+        };
+      })}
+    />
+  );
+
+  const [conditions, setConditions] = useState<SwitchVisibility["conditions"]>(
+    value.conditions ?? [],
+  );
+
+  const handleMoveUp = useCallback((idx: number) => {
+    setConditions(c => moveItemUp(idx, c) ?? c);
+  }, []);
+
+  const handleMoveDown = useCallback((idx: number) => {
+    setConditions(c => moveItemDown(idx, c) ?? c);
+  }, []);
+
+  const handleAdd = useCallback(() => {
+    setConditions(c => {
+      const id = generateID();
+      const nc: ConditionItemType = {
+        id,
+        condition: {
+          key: id,
+          operator: "===",
+          operand: true,
+          value: true,
+        },
+        title: "",
+      };
+      return c ? [...c, nc] : [nc];
+    });
+  }, []);
+
+  const handleRemove = useCallback((idx: number) => {
+    setConditions(c => removeItem(idx, c) ?? c);
+  }, []);
+
+  const handleItemUpdate = (item: ConditionItemType, index: number) => {
+    setConditions(c => {
+      const nc = [...(c ?? [])];
+      nc.splice(index, 1, item);
+      return nc;
+    });
+  };
+
+  //
+  const [selectedVisibility, setSelectedVisibility] = useState<string | undefined>(
+    value.userSettings?.selected ?? value.conditions?.[0]?.id,
+  );
+
+  const handleSelectVisibility = (id: string) => {
+    setSelectedVisibility(id);
+    postMsg({
+      action: "unselect",
+    });
+  };
+
+  const visibilityOptions = (
+    <Menu
+      items={conditions?.map((c, index) => {
+        return {
+          key: index,
+          label: (
+            <p style={{ margin: 0 }} onClick={() => handleSelectVisibility(c.id)}>
+              {c.title}
+            </p>
+          ),
+        };
+      })}
+    />
+  );
+
+  const valueRef = useRef(value);
+  const onUpdateRef = useRef<any>();
+  valueRef.current = value;
+  onUpdateRef.current = onUpdate;
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      const showConditions: [string, string][] = [["true", "false"]];
+
+      const selectedCondition = conditions.find(c => c.id === selectedVisibility)?.condition;
+      if (selectedCondition) {
+        showConditions.unshift([stringifyCondition(selectedCondition), "true"]);
+      }
+
+      onUpdateRef.current({
+        ...valueRef.current,
+        uiStyle: selectedStyle,
+        conditions,
+        userSettings: {
+          selected: selectedVisibility,
+        },
+        override: {
+          marker: {
+            show: {
+              expression: {
+                conditions: showConditions,
+              },
+            },
```

Comment:
Can you pull this in to a variable and pass that in to each layer type?
Then you can do
marker: showCondition,
polyline:showCondition,
etc

Reply:
Okey sure!